README.md

```diff
@@ -0,0 +1,400 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Unsize
+
+Unsizing relationships between two types can be defined by implementing the unsafe `Unsize` trait for a type and its target unsized type.
+These implementations describe how the metadata is derived from a source type and its metadata for its target unsized type.
+
+An example implementation of `Unsize` for `[T; N]` to `[T]` unsizing looks like the following:
+
+```rs
+// SAFETY:
+// - `Unsize::target_metadata` returns length metadata that spans the entire array exactly.
+// - `[T; N]` is a contiguous slice of `T`'s, so a pointer pointing to its data is valid to be interpreted as a pointer to a slice `[T]`.
+unsafe impl<T, const N: usize> Unsize<[T]> for [T; N] {
+    fn target_metadata((): <Self as Pointee>::Metadata) -> <[T] as Pointee>::Metadata {
+        N
+    }
+}
+```
+
+The metadata for the source type `[T; N]` is the unit type `()`, as there is no metadata for sized types.
+The implementation then just returns the length `N` from the array type, as this is the appropriate metadata for a slice produced from such an array.
+
+<!-- not the best example thats following here, but I could not think of an unsized to unsized relationship thats not trait upcasting, given custom unsized typed are not currently a thing -->
+
+Another example that does an unsized to unsized coercion is the following implementation (for trait upcasting provided by the compiler):
+
+```rs
+trait Super {}
+trait Sub: Super {}
+// SAFETY:
+// - `Unsize::target_metadata` returns a vtable provided by the vtable of the `dyn Sub` object.
+// - `dyn Super` is a super trait of `dyn Sub`, so a pointer pointing to data for a `dyn Sub` is valid to be used as a data pointer to a `dyn Super`
+unsafe impl Unsize<dyn Super> for dyn Sub {
+    unsafe fn target_metadata(metadata: <Self as Pointee>::Metadata) -> <dyn super as Pointee>::Metadata {
+        metadata.upcast()
+    }
+}
+```
+
+This is an unsizing impl required for trait upcasting, where the metadata (the vtable of the trait) of the `dyn Super` type has to be extracted from the the metadata of the `dyn Sub` trait.
+
+## CoerceUnsized
+
+To actually enable the unsizing coercion of objects, the `CoerceUnsized` trait that defines how the unsizing of the inner type occurs for a given pointer or wrapper type has to be implemented.
+An unsizing coercion is then done by the compiler implicitly when required by inserting a call to the `CoerceUnsized::coerce_unsized` function.
+
+A `CoerceUnsized` implementation has specific requirements to be valid which boil down to 2 kinds:
+
+### A non-delegating `CoerceUnsized` impl
+
+Such an impl is used for actual pointer like types, such as `&'a T` or `Arc<T>`.
+These kinds of impls are required to list a trait bound with the `Unsize` trait on the generic parameters of the pointer types.
```

Comment:
What does this mean?

Reply:
This is difficult to describe informally (the reference has it laid out better). Does 
> The implementing type and the `CoerceUnsized` target type must differ in a single generic parameter only.
> The differing generic parameters are required to then be part of a `T: Unsize<U>` bound where `T` is the relevant generic parameter of the implementing type and `U` the relevant generic parameter of the `CoerceUnsized` target type.

read better?

Reply:
I feel like the new text talks about something different. But I think I understand the new one.